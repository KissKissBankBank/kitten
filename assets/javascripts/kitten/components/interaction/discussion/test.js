import React from 'react'
import renderer from 'react-test-renderer'
import { Discussion } from './index'

describe('<Discussion />', () => {
  let component

  describe('compare with snapshot', () => {
    beforeEach(() => {
      component = renderer
        .create(
          <Discussion>
            <p className="k-u-a11y-visuallyHidden">
              Les messages de la discussion sont dans l’ordre antéchronologique
              : le premier message de la liste est le plus récent.
            </p>
            <Discussion.List>
              <Discussion.Message>
                <Discussion.Message.Avatar src="/kitten-3.jpg" />
                <Discussion.Message.Header>Claude L.</Discussion.Message.Header>
                <Discussion.Message.Content>
                  Donec sed odio dui. Nullam id dolor id nibh ultricies vehicula
                  ut id elit. Maecenas faucibus mollis interdum. Nullam id dolor
                  id nibh ultricies vehicula ut id elit. Morbi leo risus, porta
                  ac consectetur ac, vestibulum at eros. Sed posuere consectetur
                  est at lobortis.
                </Discussion.Message.Content>
              </Discussion.Message>
              <Discussion.Message>
                <Discussion.Message.Avatar src="/kitten-0.jpg" />
                <Discussion.Message.Header>Moi</Discussion.Message.Header>
                <Discussion.Message.Content>
                  Nullam quis risus eget urna mollis ornare vel eu leo. Nulla
                  vitae elit libero, a pharetra augue.
                </Discussion.Message.Content>
              </Discussion.Message>
              <Discussion.Message>
                <Discussion.Message.Avatar src="/kitten-3.jpg" />
                <Discussion.Message.Header>Claude L.</Discussion.Message.Header>
                <Discussion.Message.Content>
                  Nullam id dolor id nibh ultricies vehicula ut id elit. Cras
                  mattis consectetur purus sit amet fermentum. Integer posuere
                  erat a ante venenatis dapibus posuere velit aliquet. Praesent
                  commodo cursus magna, vel scelerisque nisl consectetur et.
                  Integer posuere erat a ante venenatis dapibus posuere velit
                  aliquet. Vestibulum id ligula porta felis euismod semper.
                  Vestibulum id ligula porta felis euismod semper.
                </Discussion.Message.Content>
              </Discussion.Message>
            </Discussion.List>
            <Discussion.Form
              onSubmit={() => {}}
              inputProps={{
                id: 'TextInput',
                'aria-label': 'Texte à envoyer',
                placeholder: 'Envoyer un message',
                onChange: () => {},
              }}
              buttonProps={{
                'aria-label': 'Envoyer',
                onClick: () => {},
              }}
            />
          </Discussion>,
        )
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
