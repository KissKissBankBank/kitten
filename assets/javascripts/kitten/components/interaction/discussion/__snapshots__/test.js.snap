// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Discussion /> compare with snapshot matches with snapshot 1`] = `
<div
  className="sc-hKwDye jnjFXk k-Discussion"
>
  <p
    className="k-u-a11y-visuallyHidden"
  >
    Les messages de la discussion sont dans l’ordre antéchronologique : le premier message de la liste est le plus récent.
  </p>
  <div
    className="k-Discussion__list"
    tabIndex="0"
  >
    <div
      className="k-Discussion__message"
    >
      <img
        alt=""
        className="k-Discussion__message__avatar"
        src="/kitten-3.jpg"
      />
      <div
        className="k-Discussion__message__header"
      >
        Claude L.
      </div>
      <div
        className="k-u-weight-400 k-Discussion__message__content"
      >
        Donec sed odio dui. Nullam id dolor id nibh ultricies vehicula ut id elit. Maecenas faucibus mollis interdum. Nullam id dolor id nibh ultricies vehicula ut id elit. Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Sed posuere consectetur est at lobortis.
      </div>
    </div>
    <div
      className="k-Discussion__message"
    >
      <img
        alt=""
        className="k-Discussion__message__avatar"
        src="/kitten-0.jpg"
      />
      <div
        className="k-Discussion__message__header"
      >
        Moi
      </div>
      <div
        className="k-u-weight-400 k-Discussion__message__content"
      >
        Nullam quis risus eget urna mollis ornare vel eu leo. Nulla vitae elit libero, a pharetra augue.
      </div>
    </div>
    <div
      className="k-Discussion__message"
    >
      <img
        alt=""
        className="k-Discussion__message__avatar"
        src="/kitten-3.jpg"
      />
      <div
        className="k-Discussion__message__header"
      >
        Claude L.
      </div>
      <div
        className="k-u-weight-400 k-Discussion__message__content"
      >
        Nullam id dolor id nibh ultricies vehicula ut id elit. Cras mattis consectetur purus sit amet fermentum. Integer posuere erat a ante venenatis dapibus posuere velit aliquet. Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Integer posuere erat a ante venenatis dapibus posuere velit aliquet. Vestibulum id ligula porta felis euismod semper. Vestibulum id ligula porta felis euismod semper.
      </div>
    </div>
  </div>
  <form
    className="k-Discussion__form"
    onSubmit={[Function]}
  >
    <div
      className="sc-gsDKAQ iICuau k-Form-TextInput__textareaContainer"
    >
      <textarea
        aria-label="Texte à envoyer"
        className="sc-bdvvtL ekFnRP k-Form-TextInput k-Discussion__form__input k-Form-TextInput--autoresize k-Form-TextInput--small k-Form-TextInput--autoResize"
        disabled={false}
        name="text"
        onChange={[Function]}
        placeholder="Envoyer un message"
        rows={1}
        style={
          Object {
            "lineHeight": "1.125rem",
          }
        }
      />
      <div
        className="k-Form-TextInput__textareaGradient"
      />
    </div>
    <button
      aria-label="Envoyer"
      className="sc-dkPtRN iiA-dfs k-Button k-Discussion__form__button k-Button--small k-Button--beryllium k-Button--fit-icon"
      onClick={[Function]}
      style={
        Object {
          "--Button-border-radius": null,
          "--Button-bullet-color": null,
        }
      }
      type="submit"
    >
      <svg
        aria-hidden={true}
        className=""
        fill="var(--color-grey-900, #222)"
        height={20}
        style={
          Object {
            "transform": null,
          }
        }
        viewBox="0 0 13 20"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        
        <path
          d="m6.3806 0 2.357 2.357 3.536 3.536c.651.65.651 1.706 0 2.357-.651.65-1.706.651-2.357 0l-1.869-1.869v11.952c0 .921-.746 1.667-1.667 1.667-.92 0-1.666-.746-1.666-1.667V6.381L2.8456 8.25c-.651.651-1.706.651-2.357 0-.651-.651-.651-1.707 0-2.357l3.535-3.536L6.3806 0Z"
        />
      </svg>
    </button>
  </form>
</div>
`;
